1. Listázzuk a <span style="color: #800000;">`kolcsonzes`</span> tábla következő oszlopait: ID, ki\_datum, vissza\_datum! 

  

a.  Az ID oszlop neve legyen Azonosító

b.  A ki\_datum oszlop neve legyen Kölcsönzés\_dátuma

c.  A vissza\_datum oszlop neve legyen Visszahozatal\_dátuma

In [2]:
SELECT ID AS 'Azonosító',
       ki_datum AS 'Kölcsönzés dátuma',
       vissza_datum AS 'Visszahozatal dátuma' -- visszahozatal dátuma
FROM kolcsonzesek

(5 rows affected)

Total execution time: 00:00:00.007

Azonosító Kölcsönzés dátuma Visszahozatal dátuma 10 2003-12-11 NULL 11 2003-12-11 NULL 12 2003-12-11 NULL 13 2006-10-01 2006-10-10 14 2006-10-02 2006-10-09

2. Melyek azok a kölcsönzések, amelyek 2005.01.01 és 2007.01.01 között történtek?

In [3]:
SELECT * -- az összes oszlop kiválasztása
FROM kolcsonzesek
WHERE ki_datum >= '2005.01.01' AND 
      ki_datum <= '2007.01.01'
-- WHERE ki_datum BETWEEN '2005.01.01' AND '2007.01.01'

(2 rows affected)

Total execution time: 00:00:00.005

ID ki mit ki_datum vissza_datum 13 1 3 2006-10-01 2006-10-10 14 2 3 2006-10-02 2006-10-09

3. Melyek azok a tagok, akiknek kedvenc stílusa a játék, és nevük a-ra végződik?

  

a.  Csak a tagok neve jelenjen meg, az oszlop neve Név legyen

In [12]:
SELECT nev AS 'NÉV'
FROM tagok
WHERE kedvencstilus = 'játék' AND
 nev LIKE '%a'

(2 rows affected)

Total execution time: 00:00:00.009

NÉV Bodó Kinga Garay Magdolna

4. Listázzuk egy oszlopban a tagok nevét, címét és kedvenc stílusát összefűzve! 

  

a.  A nevet, címet és a stílust kötőjellel válasszuk el

b.  A kötőjelek előtt és után legyen 1-1 szóköz

c.  Az oszlop neve legyen TAG

d.  Kinga keresztnevű tagok ne legyenek benne a listában

5. Mennyi lenne a dvd-k nettó ára, ha 20%-kal csökkenne? 

  

a.  Csak a dvd-k címét és a csökkentett árat jelenítsük meg

b.  A csökkentett árat nevezzük el diszkont\_ar-nak

In [15]:
SELECT nettoar * 0.8 AS 'diszkont_ar',
        cim
FROM dvd

(318 rows affected)

Total execution time: 00:00:00.010

diszkont_ar cim 3087,2000000000003 NULL 3395,2000000000003 A II. világháború története (1999) 4032,8 3D Dinosaur - adventure 4284 3D virtuális origami (2000) 4560 Az 1848-1849-iki szabadságharc története (2002) 2832 Ablak zsiráf (1999) 3392,8 Ady Endre összes művei (1999) 4254,400000000001 Afrika 3585,6000000000004 Age of Empires (2000) 3448,8 Airport Inc. 4472,8 American Rodeo : BullRider 2000 (2001) 3882,4 Angol : 8-10 éveseknek (2002) 3924,8 Angol és német nyelvtani gyakorlatok (2002) 3559,2000000000003 Angol szótanító CD-ROM 3555,2000000000003 Angol szótanító CD-ROM 2807,2000000000003 Angol szótár (1996) 4120,8 Angol-magyar, Magyar-angol hangosszótár 4510,400000000001 Angol-magyar nagyszótár (Halász Előd) 4564,8 Az áramvonalas fakanál 2712,8 Arany János: A bajusz (2002) 4017,6000000000004 Az Árpád-ház története 3131,2000000000003 Art Attack (1998) 4628,8 Art puzzle (1998) 4499,2 Az ásványok világa 4260,8 Ausztrália 3570,4 Auto 2000 - Kerékvilág 2498,4 Ázsia (2000) 3552 Biblia (1995) 4434,400000000001 Biológia '97 3055,2000000000003 Biológia (2000) 3033,6000000000004 Biológia (2001) 2785,6000000000004 Biológia : 12-14 éveseknek (2003) 3260,8 Blinky Bill : A fantasztikus léggömbkaland 3601,6000000000004 Broken Sword II. (cop. 1997) 3243,2000000000003 Budapest kalauz 4233,6 Budapest Museum 4664 Buggy 2621,6000000000004 Byte Magyarország (2002) 3308,8 Caesar III. 4849,6 Capitalism II. (2001) 4024 Capitalism Plus 2965,6000000000004 Castle Explorer 4337,6 CD Panoráma 1-4. (1998) 4208 Computer iskola (1998) 4745,6 Csizmás kandúr (1998) 3269,6000000000004 Csodálatos univerzum (1997) 3199,2000000000003 Dél-Amerika (1999) 4735,2 Dementia 1-5. (1997) 4506,400000000001 Deutsch: Anfanger (kezdő) + Telepítő 4627,2 Deutsch: Fortgeschrittene (haladó) + Installation 3501,6000000000004 Deutsch: Mittelstufe (középhaladó)+ Telepítő 4775,2 Diablo 3920 Diablo II. (2000) 4488,8 Dicső napok és hétköznapok 1848 - Digitális forrásgyűjtemény 1. 3759,2000000000003 Dreams to Reality 1-2. (1997) 4379,2 Duke Nukem 3D 2695,2000000000003 Dungeon Keeper 2. (1999) 4466,400000000001 E-Games : Collector's Edition (2001) 2606,4 Éléstár (1998) 4457,6 Élet a polgárosodó Magyarországon (2001) 3368,8 Életképek 1848 (1998) 3436 Az ember tragédiája 4267,2 Encarta World Atlas '99 1-2. (1998) 4173,6 Enemy Engaged RAH-66 4172 English 1-2. + Telepítő 2901,6000000000004 English: Intermediate (haladó) 3582,4 English by Stealth lásd: Lopva angolul 3999,2000000000003 Erdély (1996) 3131,2000000000003 Espanol: Intermedio + Telepítő 4705,6 Espanol: Principiante + Telepítő 3195,2000000000003 Észak-Amerika 4216 Európa 3096,8 The F.A. Premier League Stars 2001 3100,8 Felvételi példatár : Angol 2572 Felvételi példatár : Biológia, földrajz 4680,8 Felvételi példatár : Kémia, fizika 4764 Felvételi példatár : Magyar nyelv és irodalom 4007,2000000000003 Felvételi példatár : Matematika 2481,6000000000004 Felvételi példatár : Német 4047,2000000000003 FIFA, 2000 (1999) 3840 Fighting Force (1998) 3705,6000000000004 Final Liberation (1997) 3685,6000000000004 Fitness mesterfokon / Béres Alexandra (2001) 2987,2000000000003 Fizika : 12-14 éveseknek (2002) 3848 Fizika felkészítő az érettségi és felvételi vizsgára 4610,400000000001 Fizika világa (2001) 4720 A Föld élővilága 3144 Földrajz: 10-14 éveseknek (2002) 2538,4 Földrajz: érettségi előkészítő (2002) 3230,4 Francais 1-2. + Telepítő 3016 Francais: Débutan (kezdő) + Telepítő 4495,2 Francais: Moyen + Telepítő 2852,8 Francia-magyar szótár 3812,8 Funschool in Dreamland (1995) 3681,6000000000004 Geolexikon 4.2 : földtani adatbázis (2002) 3148,8 Gépírás 2000 4276,8 Gondolattár: aforizmák, szállóigék 4004 GTA 2 4099,2 Hadiflották I. (2002) 2812 Hadtudományi lexikon (1998) 3024,8 Hajdúböszörmény 3322,4 Halak, kétéltűek, hüllők 4324 Half-Life (1996) 2964 Half-Life Generation: Game of the Year Edition (2002) 4595,2 Half-Life Generation: Counter Strike (2002) 4582,400000000001 Half-Life Generation: Blue Shift (2002) 3577,6000000000004 Harry Potter és a

6. Melyek azok a kölcsönzések, ahol még nem hozták vissza a dvd-t?

7. Listázzuk azon diákok nevét és születési idejét, akik 1984-ben vagy 1985-ben születtek!  

  

a.  A születési dátum csak az évet, hónapot és a napot tartalmazza! 

b.  A születési dátum oszlopát nevezzük el 'Születési idő'-nek!

In [21]:
SELECT Nev,
-- CAST(Szulido AS nvarchar(11))
CAST(YEAR(Szulido) AS nvarchar(4)) +
'.' +
CAST(MONTH(Szulido) AS nvarchar(2)) +
'.' +
CAST(DAY(Szulido) AS nvarchar(2)) 
-- CAST(Szulido AS date) -- a legjobb megoldás
FROM diak
WHERE YEAR(Szulido) = 1984 OR 
      YEAR(Szulido) = 1985
-- WHERE YEAR(Szulido) IN (1984, 1985)

(3 rows affected)

Total execution time: 00:00:00.007

Nev (No column name) Kovács Imre 1984.3.2 Kapos Petra 1985.12.23 Koppány Olga 1984.2.28

8. Készítsünk lekérdezést, amely a tanulók nevét és az ebből képzett felhasználói nevét tartalmazza. 

  

a.  A felhasználói név álljon a név első kettő, illetve utolsó kettő betűjének összefűzéséből!

b.  A listát szűrjük azon tanulókra, akik teljes neve - szóközzel együtt - legalább 10 betűs!

In [23]:
SELECT Nev, 
    LEFT(Nev, 2) + RIGHT(Nev, 2) AS 'Felhasználónév'
FROM diak
WHERE LEN(Nev) >= 10

(8 rows affected)

Total execution time: 00:00:00.006

Nev Felhasználónév Kovács Imre Kore Horváth Emil Hoil Kapos Petra Kara Csóka Anna Csna Nyúl Tamás Nyás Ordasi Emma Orma Koppány Olga Koga Kozma Patrícia Koia

9. Készítsünk listát a diákok adatairól, ahol a diákok neve úgy jelenik meg, hogy először a keresztnév, majd a vezetéknév, közöttük szóközzel!  

a.  Ezen felül a születési idő három külön oszlopban jelenjen meg (év, hónap, nap). 

b.  Az oszlopokat nevezzük el értelemszerűen!

10. Készítsünk listát a munkák azonosítójáról és megnevezéséről! Egy új oszlopban azt is jelenítsük meg, hogy melyik munka hetente mennyit fizet! 

a.  Az oszlop neve legyen Heti bér, az értéket kerekítsük 1000 forintra! 

b.  A listát szűrjük azon rekordokra, ahol a kerekített heti bér 10000 Ft felett van!

In [29]:
-- óradíj * óraszám * 5 = heti bér
SELECT munkaId,
allas, 
ROUND(oradij * oraszam * 5, -3) AS 'Heti bér'
FROM munka
WHERE ROUND(oradij * oraszam * 5, -3) > 10000

(7 rows affected)

Total execution time: 00:00:00.004

munkaId allas Heti bér 5 kézbesítő 14000 7 eladó 12000 16 pincér 14000 17 pincér 14000 20 takarító 11000 21 takarító 11000 41 eladó 12000

11. A diákok számára differenciált béremelést terveznek: a középiskolások esetében 33%, egyéb esetben 17% mértékben. Készítsünk listát, amely tartalmazza a munkák azonosítóját, az állás nevét, az eredeti óradíjat és a tervezett emelt óradíjat.

  

a.  Az oszlopoknak adjuk nevet értelemszerűen!

b.  Az emelt óradíj összegét kerekítsük egészre! (Az esetlegesen megjelenő 0 tizedesjegyekkel ne foglalkozzunk)

12. A diákok számára próbaidőt írnak elő, amely a munka kezdetétől számítva 3 hónapig tart.    Jelenítsük meg a munka tábla adatait egy új oszloppal kiegészítve, amelyik a próbaidő    végének dátumát mutatja!  

  

a.  Az oszlop neve Próbaidő vége legyen! 

b.  A lista ne tartalmazza azokat a munkákat, ahol a diákok azonosítója nincs megadva!

13. Készítsünk listát a munkák azonosítóiról és a megnevezésükről! A megnevezés allas-óraszám formában jelenjen meg, pl: eladó-4. 

  

a.  Jelenítsük meg a dátumot is, de csak a évet és a hónapot, pl: 2003-07! 

b.  Ennek az oszlopnak 'Kezdés hónapja' legyen a neve! A listát szűrjük a nyári hónapokra!

14. Készítsünk listát a diákok adatairól, amely a Nev oszlop helyén a diák monogramját jeleníti meg (keresztnév, illetve vezetéknév első betűje összefűzve). 

a.  A listát szűrjük 1983.01.01 és 1987.06.01 között született diákokra!

15. A diákok a munkakezdés évének utolsó napján bónuszt kapnak, amennyiben az adott évben legalább 6 hónapot dolgoznak. 

  

a.  Jelenítsük meg, hogy az egyes munkák adatait, egy új oszlopban az éves bónusz napját is 'Bónusz dátum' néven! 

b.  A lista csak a bónuszra jogosultak adatait tartalmazza!